README.md

```diff
@@ -22,9 +22,50 @@ There are two lein sub-tasks within the v namespace intended for direct use:
 Show the effective version of the project and workspace state.
 
 ### lein v cache
-Cache the effective version of the project to a file (default is `version.clj`) in the first source directory (typically `src`).  It is possible to have the version cached to a file automatically by defining a prep task in your project:
+Cache the effective version of the project to a file (default is `version.clj`) in the first source directory (typically `src`). It is possible to have the version cached to a file automatically by defining a prep task in your project like this:
 
     :prep-tasks [["v" "cache" "src"]]
+    
+The `lein v cache` command has a variadic argument that is the following: first is the directory to output to (default to `src`), then the rest is the list of the output sufix you wish and hence format (default to `clj`). The available suffixes are: clj, cljs, cljx, cljc and edn. 
+You have two formats for file version output depending on the suffix:
+
+- Clojure source code: available via the `clj`, `cljs`, `cljc` or `cljx` option 
+- EDN data structure: available via the `edn` option
+
+#### Examples of lein v cache invocation
+
+```sh
+lein v cache src/cljs cljs
+# => output only the src/cljs/version.cljs file
+lein v cache src/clj clj edn
+# => output the src/clj/version.clj and src/clj/version.edn file
+lein v cache src edn
+# => output the src/version.edn file
+```
+
+#### Example of Clojure source code version output
```

Comment:
For all of these examples, it would be better to not use a `-DIRTY` version.

Reply:
ok, the versions used in examples have the -DIRTY removed.